OpenTelemetry (S3M) Databases apps (#1794)
SumoLogic/sumologic-documentation: docs/integrations/databases/opentelemetry/couchbase-opentelemetry.md

sidebar_label: Couchbase - OTel Collector
description: Learn about the Sumo Logic OpenTelemetry App for Couchbase.
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<img src={useBaseUrl('img/integrations/databases/couchbase-logo.png')} alt="Thumbnail icon" width="50"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>

[Couchbase](https://developer.couchbase.com/what-is-couchbase/), a modern database for enterprise applications, is a distributed document database with a powerful search engine and in-built operational and analytical capabilities. It brings the power of NoSQL to the edge and provides fast, efficient bidirectional synchronization of data between the edge and the cloud.

The Sumo Logic app for Couchbase helps you monitor activity in Couchbase. The pre-configured dashboards provide insight into the Error, events and HTTP Access pattern that help you understand your clusters.

Couchbase logs are sent to Sumo Logic through OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).

> This App has been tested with the following version of Couchbase:

Reviewer: ```suggestion
This App has been tested with Couchbase version 7.0.2
```

> Non-Kubernetes: Couchbase version: 7.0.2

Reviewer: We can delete this line.

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Couchbase-OpenTelemetry/schematics-couchbase.png')} alt="Schematics" />

## Fields creation in Sumo Logic for Couchbase

> Following are the [Fields](/docs/manage/fields/) which will be created as part of Couchbase App install if not already present:

Reviewer: Fields - embedded link does not seem to be working. It should redirect to https://help.sumologic.com/docs/manage/fields/

Author: We use relative links within docs. Link will work in your local build.

> **`db.cluster.name`** - User configured. Enter a name to identify this Couchbase cluster. This cluster name will be shown in the Sumo Logic dashboards.

Reviewer: Fields are getting rendered as a paragraph and not as seperate points.

**`db.system`** - Has a fixed value of **couchbase**.
**`deployment.environment`** - User configured. Through this Couchbase cluster is identified by the environment where it resides. For example: dev, prod or qa.

> **`sumo.datasource`** - has a fixed value of **couchbase**.

Reviewer: ```suggestion
**`sumo.datasource`** - Has a fixed value of **couchbase**.
```

If you want to use an existing OTel Collector, this step can be skipped by selecting the option of using an existing Collector.

If you want to create a new Collector please select **Add a new Collector** option.

Select the platform for which you want to install the Sumo OpenTelemetry Collector.

This will generate a command which can be executed in the machine which needs to get monitored. Once executed it will install the Sumo Logic OpenTelemetry Collector agent.

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Couchbase-OpenTelemetry/collector.png')} alt="Collector" />

### Step 2: Configure integration

In this step we will be configuring the yaml file required for Couchbase Log Collection. Path of the log file configured to capture couchbase logs needs to be given here.

The files are typically located in folder `/opt/couchbase/var/lib/couchbase/logs`.

* For Audit Log: `/opt/couchbase/var/lib/couchbase/logs/audit.log`
* For Error Log: `/opt/couchbase/var/lib/couchbase/logs/error.log`
* For Access Log: `/opt/couchbase/var/lib/couchbase/logs/http_access.log`
* For Query Log: `/opt/couchbase/var/lib/couchbase/logs/query.log`

> 

Reviewer: ```suggestion
You can add any custom fields which you want to tag along with the data ingested in sumo.
```

<Tabs
  className="unique-tabs"
  defaultValue="Linux"
  values={[
    {label: 'Linux', value: 'Linux'},
    {label: 'Windows', value: 'Windows'},
    {label: 'macOS', value: 'macOS'},
  ]}>

<TabItem value="Linux">

1.  Copy the yaml file to `/etc/otelcol-sumo/conf.d/` folder in the Couchbase instance which needs to be monitored.
2.  restart the collector using
  ```sh
  sudo systemctl restart otelcol-sumo
  ```

</TabItem>
<TabItem value="Windows">

1.  Copy the yaml file to `C:\ProgramData\Sumo Logic\OpenTelemetry Collector\config\conf.d` folder in the machine which needs to be monitored.
2.  Restart the collector using
  ```sh
  Restart-Service -Name OtelcolSumo
  ```

</TabItem>
<TabItem value="macOS">

1.  Copy the yaml file to **/etc/otelcol-sumo/conf.d/** folder in the Couchbase instance which needs to be monitored.
2.  Restart the otelcol-sumo process using the below command
  ```sh
  otelcol-sumo --config /etc/otelcol-sumo/sumologic.yaml --config "glob:/etc/otelcol-sumo/conf.d/*.yaml"
  ```

</TabItem>
</Tabs>

> This will install the app to your Sumo Logic Org. The app consists of Dashboards.

Reviewer: ```suggestion
Press **Next** .This will install the app to your Sumo Logic Org. The app consists of Dashboards.
```
